src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -26,6 +26,9 @@ public BitbucketPluginConfiguration() {
 
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) {
+        if (json.isEmpty()) {
+            setServerList(Collections.emptyList());
```

Comment:
`serverList` is initialised to `new ArrayList<>();` why do we need to set it agian?

Reply:
Because at this stage it has been initialised to the *old* value (from the saved state), but because there is no `serverList` in the json that is passed in, the `setServerList` is never called so it persists the old value. 
I think this is arguably a bug in Stapler, but we have to handle it here.